### PR TITLE
8.0 account_invoice_accrual

### PR DESCRIPTION
**Pull request by sbidoul**
This new module lets the user generate accrual move for supplier (or customer) invoices with one click on the draft invoice form.

When goods or services are received, the draft invoice is generated using standard odoo processes. If at the end of the period, the supplier invoice has not been received, the user can click "Accrue" on the invoice to generate an accrual move in the period being closed. Next period when the supplier invoice arrives, the accrual move is automatically reversed when the invoice is valided.

It is based on account_cutoff_accrual_base account mapping framework.

This PR is still WIP while we fix the last issues being discovered during testing.

This PR depends on #8

### CODE REVIEW

File: account_invoice_accrual/__openerp__.py
Comment (yvaucher) [inline account_invoice_accrual/__openerp__.py]: You can remove this from `__openerp__.py`

File: account_invoice_accrual/__openerp__.py
Comment (yvaucher) [inline account_invoice_accrual/__openerp__.py]: Please add OCA in author list

File: account_invoice_accrual/__openerp__.py
Comment (yvaucher) [inline account_invoice_accrual/__openerp__.py]: active is an alias for auto_install please remove it

File: account_invoice_accrual/wizard/account_move_reverse.py
Comment (yvaucher) [inline account_invoice_accrual/wizard/account_move_reverse.py]: You can use `()` and remove `\` but this is not mandatory

File: account_invoice_accrual/account_invoice.py
Comment (yvaucher) [inline account_invoice_accrual/account_invoice.py]: Context is not propagated

File: account_invoice_accrual/account_invoice.py
Comment (yvaucher) [inline account_invoice_accrual/account_invoice.py]: No need to check context here as we are not getting anything from it
Check to move under reverse_invoice

File: account_invoice_accrual/account_invoice.py
Comment (yvaucher) [inline account_invoice_accrual/account_invoice.py]: No need to check context

File: account_invoice_accrual/account_invoice.py
Comment (yvaucher) [inline account_invoice_accrual/account_invoice.py]: No need to check context

File: account_invoice_accrual/account_invoice.py
Comment (yvaucher) [inline account_invoice_accrual/account_invoice.py]: Please check context is dict here

File: account_invoice_accrual/account_invoice.py
Comment (yvaucher) [inline account_invoice_accrual/account_invoice.py]: missing context propagation

File: account_invoice_accrual/i18n/account_invoice_accrual.pot
Comment (yvaucher) [inline account_invoice_accrual/i18n/account_invoice_accrual.pot]: s/OpenERP/Odoo

File: account_invoice_accrual/i18n/fr.po
Comment (yvaucher) [inline account_invoice_accrual/i18n/fr.po]: s/OpenERP/Odoo

File: account_invoice_accrual/i18n/fr.po
Comment (yvaucher) [inline account_invoice_accrual/i18n/fr.po]: s/OpenERP/Odoo

File: account_invoice_accrual/wizard/account_move_accrue.py
Comment (yvaucher) [inline account_invoice_accrual/wizard/account_move_accrue.py]: s/OpenERP/Odoo

**Comment by gurneyalex** [comment]
I just restarted the travis build.

**Comment by adrienpeiffer** [comment]
@yvaucher Thanks for your review !

File: account_invoice_accrual/README.rst
Comment (yvaucher) [inline account_invoice_accrual/README.rst:4]: Can you add the license badge in README ?

see https://github.com/OCA/maintainer-tools/tree/master/template/module

**Comment by yvaucher** [comment]
@adrienpeiffer Thanks for the changes just a tiny detail and it is :+1:

File: account_invoice_accrual/README.rst
Comment (adrienpeiffer) [inline account_invoice_accrual/README.rst:4]: Of course !

Comment (adrienpeiffer) [inline account_invoice_accrual/README.rst:4]: @yvaucher it's done !

**Comment by adrienpeiffer** [comment]
Travis is green now :smile:

**Comment by yvaucher** [comment]
Thanks we need a second approval there

Maybe @sbidoul can do it ? Then we can merge

By the way I think you can remove the [WIP] in title

**Comment by sbidoul** [comment]
@yvaucher I'll do a detailed functional test soon.

File: account_invoice_accrual/account_invoice.py
Comment (sbidoul) [inline account_invoice_accrual/account_invoice.py]: What is the purpose of this accrual_move_name field?

File: account_invoice_accrual/account_invoice.py
Comment (sbidoul) [inline account_invoice_accrual/account_invoice.py]: , False not necessary

File: account_invoice_accrual/account_invoice.py
Comment (sbidoul) [inline account_invoice_accrual/account_invoice.py]: better name this reverse_accrual?

Comment (adrienpeiffer) [inline account_invoice_accrual/account_invoice.py]: Yes, I think too

File: account_invoice_accrual/account_invoice.py
Comment (sbidoul) [inline account_invoice_accrual/account_invoice.py]: I'm not sure we should reset accrual_move_id when reversing. Because if the user reverses the journal entry without going through the reversal button on the invoice, it will not be reset. So that would give a slightly different behaviour, potentially confusing.

Comment (sbidoul) [inline account_invoice_accrual/account_invoice.py]: Ah, sorry, it works because the account.move.reversal wizard is extended too. Good.

**Comment by sbidoul** [comment]
I found a scenario with an issue:
- today = 2015-05-05
- create supplier invoice dated 2015-04-15 with period 04/2015
- accrual 2015-04-30 -> ok
- change invoice period to 05/2015
- validate invoice -> the reveral move takes date 2015-06-01 and period 06/2015: NOK

When validating the invoice, the reversal move must have the same period as the invoice move (and I'd say the first day of the period).

**Comment by adrienpeiffer** [comment]
Rebased to get travis file changes

**Comment by adrienpeiffer** [comment]
- [x] Depends on #24

**Comment by sbidoul** [comment]
The build is red because it depends on #24 but is otherwise ok.

**Comment by sbidoul** [comment]
I discovered another issue when using hr_timesheet_invoice. Analytic lines linked to accrual moves are proposed for invoicing because account_move_line.invoice (which is a computed field) is False for accrual moves and therefore analytic lines are not linked to the invoice (https://github.com/odoo/odoo/blob/8.0/addons/hr_timesheet_invoice/hr_timesheet_invoice.py#L363).

This module is definitely harder to get 100% right than I expected.

**Comment by adrienpeiffer** [comment]
Work fine in production for more than 1 year

**Comment by lmignon** [comment]
Work fine in production for more than 1 year

**Review by lmignon — APPROVED** [review]

**Review by yvaucher — APPROVED** [review]

**Comment by adrienpeiffer** [comment]
Thanks @yvaucher !

**Comment by yvaucher** [comment]
@sbidoul @adrienpeiffer @lmignon This merge make the travis tests fail:
https://travis-ci.org/OCA/account-closing/jobs/162209582

Can you check ? It seems it is incompatible with `account_reversal` module

**Comment by sbidoul** [comment]
This is due to a recent change by @alexis-via in account_reversal https://github.com/OCA/account-financial-tools/pull/387 that came after the last travis run of this module PR.

**Comment by sbidoul** [comment]
I'm working on it

**Comment by sbidoul** [comment]
#38

**Pull request merged**
